In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

In [ ]:
data = pd.read_csv("wear.csv")

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15826 entries, 0 to 15825
Data columns (total 20 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   SPEAKER      15826 non-null  object 
 1   SENTENCE     15826 non-null  object 
 2   DATAID       15826 non-null  object 
 3   DOMAINID     15826 non-null  object 
 4   DOMAIN       15826 non-null  object 
 5   CATEGORY     15826 non-null  object 
 6   SPEAKERID    15826 non-null  int64  
 7   SENTENCEID   15826 non-null  int64  
 8   MAIN         15725 non-null  object 
 9   SUB          1840 non-null   object 
 10  QA           15826 non-null  object 
 11  QACNCT       240 non-null    float64
 12  MQ           7570 non-null   object 
 13  SQ           1086 non-null   object 
 14  UA           811 non-null    object 
 15  SA           6359 non-null   object 
 16  개체명          12881 non-null  object 
 17  용어사전         586 non-null    object 
 18  지식베이스        8993 non-null   object 
 19  Unna

## SPEAKER,SPEAKERID
 - SPEAKERID는 고객이면 1, 점원이면 0

In [ ]:
print("SPEAKER열의 값으로 올 수 있는 것 : ",data['SPEAKER'].unique())
print(data['SPEAKER'].value_counts())

SPEAKER열의 값으로 올 수 있는 것 :  ['고객' '점원']
고객    8381
점원    7445
Name: SPEAKER, dtype: int64


## SENTENCE
 - 15826 개의 문장중, 14094개가 유일한 문장이다.
 - '네'라는 문장이 다른 답변보다 10배 이상 많았다.
 - '네' 단어가 들어간 문장의 개수 : 2113 (생각보다 얼마 안된다?)
 - 데이터의 14900행부터 끝(15826)까지 고객의 질문만 이어진다.

In [ ]:
print("SENTENCE 열의 값으로 올 수 있는 것 : ",data['SENTENCE'].unique())
print(data['SENTENCE'].value_counts())

SENTENCE 열의 값으로 올 수 있는 것 :  ['신발은 여기 있는 게 다예요?' '네 성인이나 아동 다 있어요' '발 사이즈 몇 신으세요?' ...
 '쿠폰 몇 개 모으면 사은품 받을 수 있어요?' '쿠폰 몇 장 모아야 사은품 받을 수 있나요?'
 '사은품 받으려면 쿠폰 몇 장 있어야 돼요?']
네                             442
네 맞아요                          46
이거는 얼마예요?                      39
네 가능합니다                        39
네 알겠습니다                        37
                             ... 
네 더 필요한 건 없으세요?                 1
네, 이 제품이 봄 되면 치마식으로 내리면 돼요      1
신발 나이 드신 분이 신을 건데 조금 쿠션 있고      1
광고에 나왔던 가방들 있어요?                1
치마랑 어울릴 만한 니트 있나요?              1
Name: SENTENCE, Length: 14094, dtype: int64


In [ ]:
# '네'란 단어가 들어간 문장의 개수
ne_count = 0
for line in data.SENTENCE:
    word = line.split()
    for i in word:
        if i == '네':
            ne_count += 1
print(ne_count)

2113


## DATAID
 - 데이터를 추가한 사람의 정보: 장미연_신발_17 : 장미연씨가 신발에 대한 데이터중 17번째이다??

In [ ]:
print("DATAID열의 값으로 올 수 있는 것 : ",data['DATAID'].unique())
data['DATAID'].value_counts()

DATAID열의 값으로 올 수 있는 것 :  ['박은영_액세서리_5' '박은영_액세서리_11' '김옥연_액세서리_4' ... '김옥연_의류_4' '김옥연_의류_2' '생성']


생성            926
장미연_신발_17      90
박은영_가방_5       66
김옥연_액세서리_2     65
이정숙_가방_6       62
             ... 
장미연_가방_21       6
장미연_의류_8        6
이수연_액세서리_8      4
장미연_의류_30       4
장미연_의류_42       4
Name: DATAID, Length: 1002, dtype: int64

## DOMAINID, DOMAIN

In [ ]:
print("DOMAINID 열의 값으로 올 수 있는 것 : ",data['DOMAINID'].unique())
print(data['DOMAINID'].value_counts())

DOMAINID 열의 값으로 올 수 있는 것 :  ['B']
B    15826
Name: DOMAINID, dtype: int64


In [ ]:
print("DOMAIN 열의 값으로 올 수 있는 것 : ",data['DOMAIN'].unique())
print(data['DOMAIN'].value_counts())

DOMAIN 열의 값으로 올 수 있는 것 :  ['의복의류점']
의복의류점    15826
Name: DOMAIN, dtype: int64


## CATEGORY
 - 모든 15826개의 행이 카테고리가 있음!!!

In [ ]:
print("CATEGORY 열의 값으로 올 수 있는 것 : ",data['CATEGORY'].unique())
print(data['CATEGORY'].value_counts())

CATEGORY 열의 값으로 올 수 있는 것 :  ['신발' '가방' '액세서리' '의류']
의류      5873
신발      3802
가방      3678
액세서리    2473
Name: CATEGORY, dtype: int64


## SENTENCEID
 - SENTENCEID가 유일한 것들은 데이터 뒤쪽에 추가된 질문들의 경우

In [ ]:
print("SENTENCEID 열의 값으로 올 수 있는 것 : ",data['SENTENCEID'].unique())
print("SENTENCEID 의 unique 개수 : ",data['SENTENCEID'].nunique())
print(data['SENTENCEID'].value_counts())

SENTENCEID 열의 값으로 올 수 있는 것 :  [  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18
  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36
  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54
  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72
  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90
  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108
 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126
 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144
 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162
 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180
 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198
 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216
 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234
 235 236 237 238 239 

## MAIN
 - 총 15826개의 문장중 15725의 문장만이 해당 값을 지니고 있다.
 - 405종류가 있다.

In [ ]:
print("MAIN 열의 값으로 올 수 있는 것 : ",data['MAIN'].unique())
print("MAIN 의 unique 개수 : ",data['MAIN'].nunique())
print(data['MAIN'].value_counts())

MAIN 열의 값으로 올 수 있는 것 :  ['종류별신발제품문의요청' '착화감' '굽높이문의' '재입고문의' '소재문의' '제품가격문의' '사이즈재고문의' '사이즈문의'
 '종류별가방제품문의요청' '소재를제시한제품문의' '가방소재문의' '제품구성문의' '공휴일영업문의' '제품소재문의'
 '세트제품개별구매문의' '종류별액세서리제품문의요청' '계절상품문의' '제품문의' '매장에서착용문의' '신상품문의'
 '제품의세척방법문의' '마감시간문의' '원단문의' '디자인문의' '상품권사용문의' '제품사용방법문의' '제품하자발견시요구'
 '재고없을시입고문의' '주문제작문의' '금함량문의' '견본제품문의' '금매입문의' '제품요청' '다른매장과가격비교문의'
 '세일품목문의' '제품별색깔종류문의' '재고문의' '세일기간문의' '할인여부문의' '베스트상품문의' '현금할인문의'
 '세척서비스문의' '세척비용문의' '세트제품문의' '금시세문의' '성별선호에따른선택문의' '색상에대한문의' '가게앞주차여부'
 '결제방식문의' '수제품문의' '귀뚫어주는지문의' '선물할대상에따른제품문의' '알레르기유무문의' '오픈시간문의' '치수문의'
 '종류별의류제품문의요청' '제품의세탁방법문의' '제품추천문의' '피팅문의' '제품차이문의' '제품주문' '교환환불문의' '결제요청'
 '코디제품추천문의' '제품별추천문의' '옷감문의' '제품가격불만족' '제조사제조국문의' '브랜드종류문의' '착용후어울리는지문의'
 '가격제한시선택문의' '코디상품문의' '가격임의할인요구' '착용전후다른사이즈피팅문의' '현금결제할인문의' '인근지리와위치문의'
 '할인이유문의' '피팅사이즈주문' '할인카드문의' '매장제품구경' '유사제품추천문의' '할인율문의' '매장오픈문의' '영업시간문의'
 '소재의특성문의' '용도에따른제품추천요청' '빅사이즈제품문의' '성별제품문의' '제품위치문의' '연령에따른제품추천문의'
 '착용방법문의' '거울위치문의' '체형에따른선택문의' '매장내자체수선여부' '매장확인' '피팅룸위치문의' '장소에따른

## SUB
 - 해당 값은 15826개의 문장중 1840개만이 이 값이 있다.
 - 170종류의 값이 올 수 있다.
 - 점원의 역질문과 해당 역질문에 대한 고객의 답변에 대한 주제 이때 해당 쌍은 같은 SUB값을 가진다.
 - 전체 문장중 점원의 역질문과 그에대한 답변의 개수가 총 1840개라는 의미는 아닌것이....
 - 점원의 역질문에 대한 고객의 대답 이후 고객의 역역질문이 이어진다면, NaN값이 채워진다. 즉, Q(고)AQ(점)A(고)구조의 뒤의 Q(점)A(고)에는 값이 있지만, Q(고)AQ(점)Q(고) or  Q(고)AQ(점)AQ(고)이런식의 두번 꼬여버린 경우 이후 뒤의 Q(고)부터의 문장들은 NaN값이 채워지는 듯..
 - 정리하면, 점원의 역질문과 이에 대한 고객의 답변에 해당하는 문장은 이 값을 가진다. 역질문의 의도? 그래서 MAIN의도가 아닌 SUB의도인듯..

In [ ]:
print("SUB 열의 값으로 올 수 있는 것 : ",data['SUB'].unique())
print("SUB 의 unique 개수 : ",data['SUB'].nunique())
print(data['SUB'].value_counts())

SUB 열의 값으로 올 수 있는 것 :  [nan '사이즈' '제품문의' '제품추천' '제품착용' '종류' '대상' '제품안내' '핸드폰기종' '소재확인' '스타일'
 '알레르기' '착용' '피팅요청' '코디신발사이즈' '주문확인' '결제방법선택' '제품선택' '대상확인' '피팅추천' '제품종류'
 '용도' '체형' '연령대' '성별' '체형문의' '코디할옷확인' '사이즈추천' '제품확인' '소재특성문의' '수량' '영수증'
 '포장문의' '색상문의' '소독제여부' '제품용도' '모양종류' '제품위치' '가격대' '연령' '사이즈확인' '디자인'
 '착용대상' '색상' '굽높이' '착용감문의' '제품에대한사이즈추천' '제품문의확인' '원하는디자인문의' '주문시소요시간'
 '구매확인' '결제' '사이즈변경' '스타킹제공' 'MD추천' '현금영수증' '번호요청' '가격' '색상확인' '예약문의'
 '사이즈문의' '색상추천' '스타일추천' '할인제품' '소재' '상품권종류' '대��' '제품제시' '요청확인' '카드결제문의'
 '카드결제수수료안내' '공통' '포장지' '결제방식선택' '반지사이즈' '선물포장' '할부안내' '포장' '대기' '행사상품안내'
 '할인여부' '포인트카드' '취향' '사이즈측정' '연락처' '반지스타일' '착용여부' '귀뚫었는지여부' '확인' '소재문의'
 '수선문의' '선택제품문의' '구매후교환환불요청' '사이즈업추천' '수선기간' '브랜드문의' '할부개월수' '신고온신발포장문의'
 '부츠길이' '다른제품추천' '영수증처리' '수령방법' '원인' '할인행사안내' '하자확인' '소재재질' '쇼핑백' '색상선택'
 '할부문의' '디자인문의' '영수증확인' '제품디자인' '입고확인' '예약자' '예약제품확인' '제품상태' '수납품사이즈'
 '대상선호도' '다른색상제시' '무이자할부' '신상제품추천' '수납품' '부품소재선택' '제품상태확인' '교환제품선택'
 '교환제품금액추가' '수선정도문의' '수선비용부담' '브랜드' '이벤트' '일시불' 

## QA
 - 질문인지, 답변인지. 말하는 사람이 누구인지 상관없이.

In [ ]:
print("QA 열의 값으로 올 수 있는 것 : ",data['QA'].unique())
print("QA 의 unique 개수 : ",data['QA'].nunique())
print(data['QA'].value_counts())

QA 열의 값으로 올 수 있는 것 :  ['Q' 'A']
QA 의 unique 개수 :  2
Q    8656
A    7170
Name: QA, dtype: int64


## QACNCT
 - 전체 문장중 240개의 문장만 이 값을 지닌다.
 - ??? 정체를 모르겠다.
 - 대화중 점원의 대답(SA) 문장중 일부만이 이 값이 있다.

In [ ]:
print("QACNCT 열의 값으로 올 수 있는 것 : ",data['QACNCT'].unique())
print("QACNCT 의 unique 개수 : ",data['QACNCT'].nunique())
print(data['QACNCT'].value_counts())

QACNCT 열의 값으로 올 수 있는 것 :  [nan  1.  8.  3.  6. 16. 11.  9. 13.  5. 19. 15.  7. 17. 12. 35. 24. 33.
 49. 14. 21. 31.  2.  4.]
QACNCT 의 unique 개수 :  23
1.0     111
7.0      36
3.0      26
5.0      19
9.0       8
6.0       6
13.0      5
11.0      4
15.0      4
2.0       3
17.0      3
49.0      2
31.0      2
19.0      2
4.0       1
33.0      1
14.0      1
24.0      1
12.0      1
16.0      1
35.0      1
21.0      1
8.0       1
Name: QACNCT, dtype: int64


In [ ]:
QACNCT_data = data.dropna(subset=['QACNCT'])
QACNCT_data

SPEAKER                                          SENTENCE      DATAID DOMAINID DOMAIN CATEGORY  SPEAKERID  SENTENCEID           MAIN  SUB QA  QACNCT   MQ   SQ   UA                                                SA                  개체명 용어사전                  지식베이스  Unnamed: 19
152        점원                                           이거 괜찮아요  박은영_액세서리_5        B  의복의류점     액세서리          0           4         세일품목문의  NaN  A     1.0  NaN  NaN  NaN                                           이거 괜찮아요                   이거  NaN                    NaN          NaN
177        점원                                 18케이로 하시면 49만 원이요  박은영_액세서리_6        B  의복의류점     액세서리          0          11         제품가격문의  NaN  A     8.0  NaN  NaN  NaN                                 18케이로 하시면 49만 원이요          18케이, 49만 원  NaN      18케이/순도, 49만 원/금액          NaN
195        점원                     만 오천 원대부터 시작해서 이만 오천 원까지 있습니다  박은영_액세서리_7        B  의복의류점     액세서리          0           6         제품가격문의  NaN  A     3.0  NaN  NaN  NaN                     만 오천 원대부터 시작해서 이만 오천 원까지 있습니다  만 오천 원, 시작, 이만 오천 원  NaN  만 오천 원/금액, 이만 오천 원/금액          NaN
231        점원                                      이게 두 세 돈짜리예요  박은영_액세서리_9        B  의복의류점     액세서리          0           4  종류별액세서리제품문의요청  NaN  A     1.0  NaN  NaN  NaN                                      이게 두 세 돈짜리예요                두 세 돈  NaN               두 세 돈/중량          NaN
251        점원                                이렇게 심플한 거 많이 나가거든요  허정화_액세서리_1        B  의복의류점     액세서리          0           6    성별선호에따른선택문의  NaN  A     1.0  NaN  NaN  NaN                                이렇게 심플한 거 많이 나가거든요                   심플  NaN                 심플/스타일          NaN
...       ...                                               ...         ...      ...    ...      ...        ...         ...            ...  ... ..     ...  ...  ...  ...                                               ...                  ...  ...                    ...          ...
14579      점원                                          이런 옷 좋아요    조보월_의류_2        B  의복의류점       의류          0          20         시착가능문의  NaN  A    15.0  NaN  NaN  NaN                                          이런 옷 좋아요                    옷  NaN                    NaN          NaN
14781      점원                             70대시면 이 색이 더 좋지 않을까요?    홍은민_의류_5        B  의복의류점       의류          0          12           제품비교  NaN  A     9.0  NaN  NaN  NaN                             70대시면 이 색이 더 좋지 않을까요?                70대,색  NaN                70대/나이대          NaN
14811      점원                             원래 66이면 66사이즈 입으셔야 해요    홍은민_의류_6        B  의복의류점       의류          0          20          사이즈문의  NaN  A    17.0  NaN  NaN  NaN                             원래 66이면 66사이즈 입으셔야 해요             66,66사이즈  NaN              66사이즈/사이즈          NaN
14849      점원                                                 네    김옥연_의류_3        B  의복의류점       의류          0           6         계절상품문의  NaN  A     3.0  NaN  NaN  NaN                                                 네                  NaN  NaN                    NaN          NaN
14855      점원  검정이 아무리 기본이어도, 지금 손님 피부톤이랑 맞지 않아요. 베이지 색이 잘 어울려요    김옥연_의류_3        B  의복의류점       의류          0          12    원하는제품재고여부확인  NaN  A     7.0  NaN  NaN  NaN  검정이 아무리 기본이어도, 지금 손님 피부톤이랑 맞지 않아요. 베이지 색이 잘 어울려요      점정,기본,피부톤,베이지 색  NaN           검정/색,베이지 색/색          NaN

[240 rows x 20 columns]

## MQ
 - 7570개의 문장만이 이 값을 지닌다.
 - 사용자(손님) 질문(Main Question)의 문장에만 있는 값. 고객의 질문 문장이 그대로 들어간다.

사용자(손님) 질문(Main Question), 메인 질문에 추가적으로 필요한 시스템(점원)의 서브 질문(Sub Question), 서브 질문에 대한 사용자(손님) 응답(User Answer), 시스템(점원) 최종 응답(System Answer)로 구분

In [ ]:
print("MQ 열의 값으로 올 수 있는 것 : ",data['MQ'].unique())
print("MQ 의 unique 개수 : ",data['MQ'].nunique())
print(data['MQ'].value_counts())

MQ 열의 값으로 올 수 있는 것 :  ['신발은 여기 있는 게 다예요?' nan '굽 좀 높은 거 없나요?' ... '쿠폰 몇 개 모으면 사은품 받을 수 있어요?'
 '쿠폰 몇 장 모아야 사은품 받을 수 있나요?' '사은품 받으려면 쿠폰 몇 장 있어야 돼요?']
MQ 의 unique 개수 :  7056
이거는 얼마예요?           39
이 옷은 얼마예요?          18
이런 거는 얼마예요?         13
얼마인가요?              11
얼마예요?               11
                    ..
환불이나 교환은 되나요?        1
등산복도 드라이 해야 하나요?     1
요거는 지금 얼마 정도에요?      1
이거는 다 겨울용이다 그죠?      1
이건 버버리다 그죠?          1
Name: MQ, Length: 7056, dtype: int64


## SQ
 - 1086개의 문장만이 이 값을 지닌다.
 - MQ에 추가적으로 필요한 시스템(점원)의 서브 질문(Sub Question). 점원의 역질문. 마찬가지로 점원의 질문문장이 그대로 들어간다.

In [ ]:
print("SQ 열의 값으로 올 수 있는 것 : ",data['SQ'].unique())
print("SQ 의 unique 개수 : ",data['SQ'].nunique())
print(data['SQ'].value_counts())

SQ 열의 값으로 올 수 있는 것 :  [nan '발 사이즈 몇 신으세요?' '편하게 신을 수 있는 거 찾으세요?' '이런 건 어떠세요? 이런 거도 신발 무척 편하거든요'
 '이거 한 번 신어보세요' '보석은 싫으세요?' '악세서리 어떤 거 찾으세요?' '고객님이 하실 거예요?'
 '이거 샘플이 14케이인데 뭘로 봐드릴까요?' '굵은 거 찾으시면 이런 건 어떠세요?' '기종이 어떻게 되죠? '
 '순금으로 보여드릴까요?' '누가 하실 건데요?' '어떤 디자인 찾으세요?' '원터치로 보여드릴까요?' '알레르기 있으세요?'
 '따님요?' '누가 끼시는 거죠?' '한 쪽만 먼저 착용 해보시겠어요?' '시계 종류 어떤 거요?' '알레르기 있으신가요?'
 '네 사이즈가 어떻게 되세요?' '이런 거 말씀하시는 건가요?' '이것도 괜찮지 않아요?'
 '조끼에다가 바지만 입으셔도 예쁜데 한 번 입어보시겠어요?' '색상으로 봤을 때는 이 색상이 제일 어울려 보이는데요?'
 '뭐를 찾으십니까?' '남방 같은 거 어떠세요?' '한 번 입어 보세요' '찾으시는 거 있어요?'
 '손님 66 입으세요? 55 입으세요?' '신발 35사이즈 신으세요?' '그거 담아드려요?'
 '이 제품은 할인 상품이기 때문에 현금으로 계산해 주세요' '그럼 이번만 카드로 계산해드릴게요' '이런 꽃무늬는 어떠세요?'
 '어떤 스타일 찾으세요?' '아기자기한 거면 이런 거 어떠세요?' '반팔 티 찾으시는 건가요?' '긴 팔로 찾으세요?'
 '누구 선물하시게요?' '체격 크세요?' '사이즈 몇 짜리 보시는데요?' '이게 신축성이 있어서 괜찮을 것 같아요'
 '기모도 돼 있는 거 찾으시는 거예요?' '몇 호 입으시는데요?' '발열 내의 이거는 100도 큰데 어떠세요?'
 '뭐 찾으시는데요?' '저 재킷 이쁜데 어떠세요?' '사이즈가 어떻게 돼요?' '손님 사이즈가 어떻게 되세요?'
 '한 번 입어보세요' '이런 하얀 거는 어떠세요?' '이거 입어 보실거예요?' '인디언 핑크와 조끼 한 번 입어보세요'
 '어디서

## UA
 - 811개의 문장만이 이 값을 지닌다.
 - SQ에 대한 사용자(손님) 응답(User Answer)

In [ ]:
print("UA 열의 값으로 올 수 있는 것 : ",data['UA'].unique())
print("UA 의 unique 개수 : ",data['UA'].nunique())
print(data['UA'].value_counts())

UA 열의 값으로 올 수 있는 것 :  [nan '230이요' '네 봄이니까 편하게 신을 수 있는 거' '좀 크네' '그건 흥미없고' '스카프 같은 거' '네 '
 '18케이로' '너무 사치스러운데 ' '엘지인데' '제가 할 거예요' '그냥 기본 디자인 찾고 있어요' '원터치로 좀 붙는 걸로'
 '저는 은침해도 한 시간 지나면 심해져서요' '네' '제가요' '고민 좀 해볼게요' '메탈 시계요' '티타늄도 알러지가 생겨버려요'
 '32' '네 저도 그 색이 제일 좋은 거 같아요' '이 조끼 안에 입을만한 옷이요' '점잖아 보인다' '사이즈가 맞기는 한데요'
 '현찰은 없어요' '아기자기한 거 좋아하거든요' '그런 거 집에 많아요' '아뇨' '네 조금 크기는 한데' '105 정도요'
 '따뜻하겠네요' '네 기모요' '105 사이즈' '봄에 입을 정장 같은 거 찾는데 ' '제일 작은 거 스몰로' '44 입는데 '
 '좀 더 둘러볼게요' '사이즈가 좀 넉넉하면 좋겠어요' '너무 고상하네' '이거는 내가 너무 뚱뚱해 보여서 안되겠네요'
 '요번에 조금 격식 차릴 일이 있어갖고 좀 이쁜 거 있나 싶어갖고 들어왔거든요' '66이요' '제가 입을 거예요' '77이요'
 '내가 입을 게 아인데' '우리 딸내미가요' '네 왜소한 55 정도' '키가 좀 커요' '고등학교 2학년이요' '체격은 보통이에요'
 '나는 좀 밝은 색을 사주고 싶은데' '애가 뚱뚱해요' '더 많이 뚱뚱하지'
 '어깨가 좁으셔 가지고 한 95정도인데 엉덩이가 커가지고' '네 80대' '여자분이요' '그건 별로 안좋아하실 거 같아요'
 '105 정도' '그럴게요' '네 입어볼게요' '아니에요' '우리애가 입을 거 사달라고 해서요'
 '체형이 위는 조금 말랐는데 하체는 굵거든요' '약간 이런 색깔인데' '네 코트' '네 백이예요' '네 그럼 그렇게 주세요'
 '이거는 소재가 광택이 좀 나네요' '아이보리 이거 이쁘네요' '네 여학생이에요'
 '구십오 백은 해야 되겠죠 어른들 안에 뭐 입고 해야되니까' '연세는 

## SA
 - 6359개의 문장만이 이 값을 지닌다.
 - 시스템(점원) 최종 응답(System Answer)
 - 점원의 답변에 해당하는 문장은 전부 이 값이 있다.
 - 8656(Q) - 7570(MQ) = 1086 = SQ : 점원의 질문 
 - 7170(A) - 6359(SA) = 811 = UA : 고객의 답변 

In [ ]:
print("SA 열의 값으로 올 수 있는 것 : ",data['SA'].unique())
print("SA 의 unique 개수 : ",data['SA'].nunique())
print(data['SA'].value_counts())

SA 열의 값으로 올 수 있는 것 :  [nan '네 성인이나 아동 다 있어요' '봄 상품은 아직 어른 제품이 많이 안나왔습니다' ... '그 옷은 십이만 팔천 원이에요'
 '지금 봄옷 입기 괜찮아요' '요즘 파스텔 톤이 유행이에요']
SA 의 unique 개수 :  5297
네                    407
네 맞아요                 43
네 가능합니다               39
네 알겠습니다               36
네 맞습니다                28
                    ... 
네. 감사합니다               1
원래 이런 스타일로 입는거예요       1
그거는 6만 원입니다            1
해보셔도 돼요                1
네 이쪽에 가지고 오셔서 푸세요      1
Name: SA, Length: 5297, dtype: int64


 16  개체명          12881 non-null  object 
 17  용어사전         586 non-null    object 
 18  지식베이스        8993 non-null   object 

## 개체명
 - 12881개의 문장만 이 값을 가지고 있다.

In [ ]:
print("개체명 열의 값으로 올 수 있는 것 : ",data['개체명'].unique())
print("개체명 의 unique 개수 : ",data['개체명'].nunique())
print(data['개체명'].value_counts())

개체명 열의 값으로 올 수 있는 것 :  ['신발, 여기' '성인, 아동' '발, 사이즈' ... '옷,십이만 팔천 원' '유행,색깔' '파스텔 톤,유행']
개체명 의 unique 개수 :  8061
사이즈              181
옷                127
얼마               114
가방               114
신발                94
                ... 
육육사이즈              1
구두, 처음, 볼          1
지금, 전체, 세일 기간      1
줄무늬,옷              1
옷,현금가,십 오만 원       1
Name: 개체명, Length: 8061, dtype: int64


## 용어사전
 - 586개의 문장만이 이 값을 지닌다.
 - 다른 단어이이지만, 의미가 같은 용어들을 모아놓은 것인듯..
 - 뒤의 단어가 좀더 표준적이므로, 전처리할때 쓰이면 될수도.. ex. '디씨' or '디시'라는 단어가 질문에 있을때 -> '할인'으로 바꿔서 인코딩 하는 이런 표준화 작업도 성능향상에 도움이 될듯하다.

In [ ]:
print("용어사전 열의 값으로 올 수 있는 것 : ",data['용어사전'].unique())
print("용어사전 의 unique 개수 : ",data['용어사전'].nunique())
print(data['용어사전'].value_counts())

용어사전 열의 값으로 올 수 있는 것 :  [nan '디씨/할인' '35/235' '현찰/현금' 'dc/할인' '마이/자켓' '72/172' '디피/진열' '후드/후드티'
 '신상/신제품' '쓰리 엑스라지/XXXL' '세타/스웨터' '디시/할인' '삼십오/이백삼십오' '사십오/이백사십오'
 '삼 점 오/3.5' '사십/이백사십' '사십/이백사십, 사십오/이백사십오' '삼십/이백삼십, 삼십오/이백삼십오'
 '30/230, 35/235' '앵글/앵글부츠' '35/235, 40/240' '30/230' '25/225, 50/250'
 '30230' '30/230, 40/240' '40/240' '45/245' '치수/사이즈' '65/265'
 '35/235, 30/230' '피/가죽' '홈플/홈플러스 문화상품권, 신세계/신세계 문화상품권' 'DC/할인' '내자/레자'
 '200/200만 원' '사십/사십만원' '10프로/10%' '데일리/평소' '75000/75000원' 'a/s/as'
 '이어링/귀걸이' '오만 원/50000원, 삼만 오천 원/35000원' '삼만 삼천 원/33000원' '이만 오천 원/25000원'
 '오만 오천 원/55000원' '17만 8천 원/178000원' '백반지/백금반지' '925/92.5%' '신상품/신제품'
 '색깔/색상' '35만 9천 원/359000원' '260짜리/260' '10퍼센트/10%' '캠퍼스/캔버스화'
 '6만 5천 원/65000원' 'A/S/AS' '데일리/날마다' '십 단위/10단위' '컬러/색상' '20만 원/200000원'
 '30프로/30%' '30만 원/300000원' '2만 원/20000원' '크기/사이즈' '5만 9천 원/59000원'
 'DP/진열' '5만 5천 원/55000원' '12만 원/120000원' '새로 나온 가방/신제품' '성함/이름'
 '10만 원/100000원' '20프로/20%' '20 퍼센트/20%' '90만 원/900000원' '신상 제품/신제품'
 '디스카운트/할인' '신상 제

## 지식베이스
 - 8993개의 문장만이 이 값을 지닌다.
 - 해당 단어가 어떤 큰 범주에 속하는지를 나타내는것 같다.

In [ ]:
print("지식베이스 열의 값으로 올 수 있는 것 : ",data['지식베이스'].unique())
print("지식베이스 의 unique 개수 : ",data['지식베이스'].nunique())
print(data['지식베이스'].value_counts())

지식베이스 열의 값으로 올 수 있는 것 :  ['신발/제품' '성인/대상, 아동/대상' nan ... '스카프/의류,팔만 오천원/가격' '십이만 팔천 원/가격' '파스텔 톤/색']
지식베이스 의 unique 개수 :  4626
가방/제품              130
가방/생활용품             99
한 번/횟수              90
가죽/재질               88
신발/제품               86
                  ... 
여자친구/대상, 목걸이/제품      1
레깅스/액세서리             1
남성용 운동화/신발 종류        1
피브이씨/재질              1
30만 원 대/가격대          1
Name: 지식베이스, Length: 4626, dtype: int64


## Unnamed: 19
 - 1개의 문장만이 이 값을 지닌다.
 - 맨 처음 문장만이 이 값을 지니므로, 별 의미는 없는 듯하다.